src/main/java/fi/aalto/cs/apluscourses/model/SubmissionResult.java

```diff
@@ -1,29 +1,62 @@
 package fi.aalto.cs.apluscourses.model;
 
 import org.jetbrains.annotations.NotNull;
+import org.json.JSONObject;
 
 public class SubmissionResult {
 
+  public enum Status {
```

Comment:
I think you could attach values to `Enum` fields as well (Java allows [that](https://www.baeldung.com/java-enum-values#adding-constructor)), and then you can compare with them like `Status.GRADED.value() .equals(...)` wo having plain `String` (lines 44-47)

Reply:
Good point, in this case the string comparisons are just used for the string value in the end point response, elsewhere the enums are compared directly.